src/frontend/src/components/nav/MainMenu.tsx

```diff
@@ -4,46 +4,60 @@ import {
   IconChevronDown,
   IconHeart,
   IconLogout,
+  IconPlugConnected,
   IconSettings,
-  IconUserCircle
+  IconUserCircle,
+  IconUserCog
 } from '@tabler/icons-react';
+import { useContext } from 'react';
 import { Link } from 'react-router-dom';
 
+import { UserContext } from '../../contexts/UserContext';
 import { doClassicLogout } from '../../functions/auth';
 import { InvenTreeStyle } from '../../globalStyle';
-import { useApiState } from '../../states/ApiState';
-import { PlaceholderPill } from '../items/Placeholder';
 
 export function MainMenu() {
   const { classes, theme } = InvenTreeStyle();
-  const [username] = useApiState((state) => [state.user?.name]);
+
+  const user = useContext(UserContext);
+
   return (
     <Menu width={260} position="bottom-end">
       <Menu.Target>
         <UnstyledButton className={classes.layoutHeaderUser}>
           <Group spacing={7}>
-            <Text weight={500} size="sm" sx={{ lineHeight: 1 }} mr={3}>
-              {username ? (
-                username
-              ) : (
-                <Skeleton height={20} width={40} radius={theme.defaultRadius} />
-              )}
-            </Text>
```

Comment:
Question: Why is the username removed? How is the user expected to know which user they are currently are using

Reply:
Oversight on my part, changed that but did not mean to commit